components/camera/fake/camera.go

```diff
@@ -241,7 +284,160 @@ func (c *Camera) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, err
+		PayloadType:    96,
+		PayloadMaxSize: webrtcPayloadMaxSize,
+	}
+
+	if err := encoder.Init(); err != nil {
+		return rtppassthrough.NilSubscription, err
+	}
+
+	c.subAndCBByID[sub.ID()] = subAndCB{
+		cb:          packetsCB,
+		sub:         sub,
+		subCancelFn: subCancelFn,
+	}
+	sub.Start()
+	g.Success()
+	return rtppassthrough.Subscription{ID: sub.ID(), Context: subCtx}, nil
+}
+
+// Unsubscribe terminates the subscription.
+func (c *Camera) Unsubscribe(ctx context.Context, id rtppassthrough.SubscriptionID) error {
+	if !c.RTPPassthrough {
+		return ErrRTPPassthroughNotEnabled
+	}
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	subAndCB, ok := c.subAndCBByID[id]
+	if !ok {
+		return errors.New("id not found")
+	}
+	subAndCB.sub.Close()
+	delete(c.subAndCBByID, id)
+	subAndCB.subCancelFn()
+	return nil
+}
+
+func (c *Camera) startPassthrough() error {
+	forma := &format.H264{
+		PayloadTyp:        96,
+		PacketizationMode: 1,
+		SPS: []uint8{
+			0x67, 0x64, 0x0, 0x15, 0xac, 0xb2, 0x3, 0xc1, 0x1f, 0xd6,
+			0x2, 0xdc, 0x8, 0x8, 0x16, 0x94, 0x0, 0x0, 0x3, 0x0, 0x4, 0x0, 0x0, 0x3, 0x0,
+			0xf0, 0x3c, 0x58, 0xb9, 0x20,
+		},
+		PPS: []uint8{0x68, 0xeb, 0xc3, 0xcb, 0x22, 0xc0},
+	}
+	rtpEnc, err := forma.CreateEncoder()
+	if err != nil {
+		c.logger.Error(err.Error())
+		return err
+	}
+	rtpTime := &rtptime.Encoder{ClockRate: forma.ClockRate()}
+	err = rtpTime.Initialize()
+	if err != nil {
+		c.logger.Error(err.Error())
+		return err
+	}
+	start := time.Now()
+	b, err := base64.StdEncoding.DecodeString(worldH264Base64)
+	if err != nil {
+		c.logger.Error(err.Error())
+		return err
+	}
+	aus, err := h264.AnnexBUnmarshal(b)
+	if err != nil {
+		c.logger.Error(err.Error())
+		return err
+	}
+	f := func() {
+		defer c.unsubscribeAll()
+		ticker := time.NewTicker(200 * time.Millisecond)
+		defer ticker.Stop()
+		for range ticker.C {
+			if c.ctx.Err() != nil {
+				return
+			}
+
+			pkts, err := rtpEnc.Encode(aus)
+			if err != nil {
+				c.logger.Error(err)
+				return
+			}
+
+			ts := rtpTime.Encode(time.Since(start))
+			for _, pkt := range pkts {
+				pkt.Timestamp = ts
+			}
+
+			// get current timestamp
+			c.mu.RLock()
+			for _, subAndCB := range c.subAndCBByID {
+				// write packets
+				if err := subAndCB.sub.Publish(func() { subAndCB.cb(pkts) }); err != nil {
+					c.logger.Warn("Publish err: %s", err.Error())
+				}
+			}
+			c.mu.RUnlock()
+		}
+	}
+	c.activeBackgroundWorkers.Add(1)
+	utils.ManagedGo(f, c.activeBackgroundWorkers.Done)
+	return nil
+}
+
+func (c *Camera) unsubscribeAll() {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	for id, subAndCB := range c.subAndCBByID {
+		subAndCB.sub.Close()
+		delete(c.subAndCBByID, id)
+		subAndCB.subCancelFn()
+	}
+}
+
 // Close does nothing.
 func (c *Camera) Close(ctx context.Context) error {
+	c.cancelFn()
+	c.activeBackgroundWorkers.Wait()
 	return nil
 }
+
+// NOTE: (Nick S) This is a single frame of h264. This makes the implementation simple but results in the video appearing as an image.
```

Comment:
I recommend shoving this into a separate file under `camera/fake`.

Reply:
done